appinventor/appengine/src/com/google/appinventor/server/project/CommonProjectService.java

```diff
@@ -387,4 +387,12 @@ public TextFile importMedia(String userId, long projectId, String urlString, boo
       }
     }
   }
+
+  public long renameFile(String userId, long projectId, String oldFileId, String newFileId) {
+    storageIo.addSourceFilesToProject(userId, projectId, false, newFileId);
+    byte[] content = storageIo.downloadRawFile(userId, projectId, oldFileId);
+    long modDate = storageIo.uploadRawFileForce(projectId, newFileId, userId, content);
+    storageIo.removeSourceFilesFromProject(userId, projectId, true, oldFileId);
```

Comment:
We may want to verify we can read the file back from storage before deleting the old copy.